src/rgw/rgw_admin.cc

```diff
@@ -1260,6 +1291,50 @@ static bool dump_string(const char *field_name, bufferlist& bl, Formatter *f)
   return true;
 }
 
+void set_qos_info(RGWQoSInfo& qos, OPT opt_cmd, int64_t max_read_ops, int64_t max_write_ops,
+                    int64_t max_read_bytes, int64_t max_write_bytes,
+                    bool have_max_read_ops, bool have_max_write_ops,
+                    bool have_max_read_bytes, bool have_max_write_bytes)
+{
+  switch (opt_cmd) {
+    case OPT::RATELIMIT_ENABLE:
+    case OPT::GLOBAL_RATELIMIT_ENABLE:
+      qos.enabled = true;
+
+      // falling through on purpose
+
+    case OPT::RATELIMIT_SET:
+    case OPT::GLOBAL_RATELIMIT_SET:
+      if (have_max_read_ops) {
+        if (max_read_ops > -1) {
```

Comment:
is -1 a special value that the use can pass to the commandline, or is this a default value?